presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/functionNamespace/NativeFunctionNamespaceManager.java

```diff
@@ -0,0 +1,484 @@
+        }
+        return functionHandle;
+    }
+
+    /*
+        Todo: Improve the handling of parameter type differentiation in native execution.
+        Currently, we lack support for correctly identifying the parameterKind, specifically between TYPE and VARIABLE,
+        in native execution. The following utility functions help bridge this gap by parsing the type signature and verifying whether its base
+        and parameters are of a supported type. The valid types list are non - parametric types that Presto supports.
+     */
+
+    public static TypeSignature convertApplicableTypeToVariable(TypeSignature typeSignature, List<Type> types)
+    {
+        List<TypeSignature> typeSignaturesList = convertApplicableTypeToVariable(ImmutableList.of(typeSignature), types);
+        checkArgument(!typeSignaturesList.isEmpty(), "Type signature list is empty for : " + typeSignature);
+        return typeSignaturesList.get(0);
+    }
+
+    public static List<TypeSignature> convertApplicableTypeToVariable(List<TypeSignature> typeSignatures, List<Type> types)
+    {
+        List<TypeSignature> newTypeSignaturesList = new ArrayList<>();
+        for (TypeSignature typeSignature : typeSignatures) {
+            if (!typeSignature.getParameters().isEmpty()) {
+                TypeSignature newTypeSignature =
+                        new TypeSignature(
+                                typeSignature.getBase(),
+                                getTypeSignatureParameters(
+                                        typeSignature,
+                                        typeSignature.getParameters(),
+                                        types));
+                newTypeSignaturesList.add(newTypeSignature);
+            }
+            else {
+                newTypeSignaturesList.add(typeSignature);
+            }
+        }
+        return newTypeSignaturesList;
+    }
+
+    private static List<TypeSignatureParameter> getTypeSignatureParameters(
+            TypeSignature typeSignature,
+            List<TypeSignatureParameter> typeSignatureParameterList,
+            List<Type> types)
+    {
+        List<TypeSignatureParameter> newParameterTypeList = new ArrayList<>();
+        for (TypeSignatureParameter parameter : typeSignatureParameterList) {
+            if (parameter.isLongLiteral()) {
+                newParameterTypeList.add(parameter);
+                continue;
+            }
+
+            boolean isNamedTypeSignature = parameter.isNamedTypeSignature();
+            TypeSignature parameterTypeSignature;
+            // If it's a named type signatures only in the case of row signature types.
+            if (isNamedTypeSignature) {
+                parameterTypeSignature = parameter.getNamedTypeSignature().getTypeSignature();
+            }
+            else {
+                parameterTypeSignature = parameter.getTypeSignature();
+            }
+
+            if (parameterTypeSignature.getParameters().isEmpty()) {
```

Comment:
Let's very clearly call out the hack, where it is needed and why.  Let's prefix the comment with `// HACK:` so it's obvious.

Reply:
It seems like this is several methods long, so let's add an end comment that says when the hack is over.